File: final_qualifying_work/Projects/data_provider_app/app/src/main/java/com/example/data_provider_app/dto/CarDto.kt
package com.example.data_provider_app.dto

import java.time.LocalDateTime

data class CarDto(
    val carId: UInt,
    val createdAt: LocalDateTime,
    val personId: UInt,
    val vinNumber: String,
    val stateNumber: String?,
    val modelName: String,
    val brandName: String,
    val bodyName: String,
    val releaseYear: UShort,
    val gearboxName: String,
    val driveName: String,
    val enginePowerHP: UShort,
    val enginePowerKW: Float,
    val engineCapacityL: Float,
    val tankCapacityL: UByte,
    val fuelTypeName: String,
    val vehicleWeightKG: UShort,
    val photoId: UInt
)

File: final_qualifying_work/Projects/server/Models/Dtos/CarDto.cs
using server.Models.Entities;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;

namespace server.Models.Dtos
{
    public class CarDto
    {
        public uint CarId { get; set; }
        public uint PersonId { get; set; }
        public string VINNumber { get; set; } = null!;
        public string? StateNumber { get; set; }
        public string ModelName { get; set; } = null!;
        public string BrandName { get; set; } = null!;
        public string BodyName { get; set; } = null!;
        public ushort ReleaseYear { get; set; }
        public string GearboxName { get; set; } = null!;
        public string DriveName { get; set; } = null!;
        public ushort EnginePowerHP { get; set; }
        public float EnginePowerKW { get; set; }
        public string EngineTypeName { get; set; } = null!;
        public float EngineCapacityL { get; set; }
        public byte TankCapacityL { get; set; }
        public string FuelTypeName { get; set; } = null!;
        public ushort VehicleWeightKG { get; set; }
        public bool IsArchived { get; set; }
    }
}
